gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -924,6 +935,212 @@ public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOExce
     }
   }
 
+  /** See {@link GoogleCloudStorage#deleteFolders(List)} for details about expected behavior. */
+  @Override
+  public void deleteFolders(List<FolderInfo> folders) throws IOException {
+    // Gather exceptions to wrap in a composite exception at the end.
+    final KeySetView<IOException, Boolean> innerExceptions = ConcurrentHashMap.newKeySet();
+    String traceContext = String.format("batchFolderDelete(size=%s)", folders.size());
+    try (ITraceOperation to = TraceOperation.addToExistingTrace(traceContext)) {
+
+      // threads for parallel delete of folder resources
+      BatchExecutor executor = new BatchExecutor(storageOptions.getBatchThreads());
+
+      // Map to store number of children for each parent object
+      HashMap<String, Long> countOfChildren = new HashMap<>();
```

Comment:
Do we have to synchronize access to this since this can be updated concurrently?